addons/hr/data/hr_demo.xml

```diff
@@ -211,6 +211,12 @@ If you have development competencies, we can propose you specific traineeships</
           <field name="create_date">2010-01-01 00:00:00</field>
       </record>
 
+      <record id="work_contact_al" model="res.partner">
```

Comment:
Well this is because it is demo data associated with the hr module. As such, for the uninstallation process to work correctly (and remove the data) there needs to be an `xml_id` associated with this data.

Simply adding the data to the xml demo data is in my opinion the cleanest way to accomplish this. Alternatively the `xml_id` could be created on the fly in the code, for example whenever the employee also has an associated `xml_id`. However, to me this seems more convoluted and bug-prone.

If no `xml_id` exists for these `res.partner` records, they will stay behind in the database after uninstalling the `hr` module (all the details are in the associated opw ticket).

Reply:
Ok yes, that makes a lot of sense :)